tests/views_integration/Tribe/Events/Views/V2/Partials/Components/Ical_LinkTest.php

```diff
@@ -94,6 +101,27 @@ public function test_render_with_all() {
 		);
 	}
 
+	public function test_render_single_with_all() {
+		global $wp_query;
+		$wp_query->is_single = true;
+
+		$event_id = tribe_events()->set_args( [
+			                                      'start_date' => '2018-07-01 10am',
+			                                      'timezone'   => 'Europe/Paris',
+			                                      'duration'   => 3 * HOUR_IN_SECONDS,
+			                                      'title'      => 'Test Event - 2018-07-01 11am',
+			                                      'status'     => 'publish',
+		                                      ] )->create();
+		$GLOBALS['post'] = get_post( $event_id );
```

Comment:
If we add another test with invalid event ID that should confirm the fatal scenario discussed above.